### ISSUE TITLE
Adaptation to other games

### ISSUE THREAD

**Issue by MarkusMontenegro** [tensorflow/minigo#949]
Hi,
I'm not sure if this is the right place to ask this question, but anyways:
I'm trying to adapt this project to play a different game: It's about walking in a map containing walls and collecting more points than the opponent. 
By changing only the file go.py i managed to change rules so that the game is played, terminated and scored as intended.
But im struggeling with learning new models for this game.
In features.py i added 2 features planes to all sets of features, one representing walls and one representing the points that need to be collected.
As far as I understand the azg paper, this should be enough to learn the game by selfplay.
Still, I don't see any results when doing so.
The longest training I have done so far were 15 iterations of playing and learning a new model, and both players still seem to move randomly.
Should my approach (theoretically) be working, or am I missing something important?

**Comment by brilee** [comment]
My first inclination would just be to suspect a bug. If you push your code, I can take a quick skim over it.

**Comment by amj** [comment]
Hi!  It's really hard to tell if something is "slightly better than random", so it's tough to know how fast to expect the agents to improve.  The best advice i can give is to try and simplify the problem as much as possible until you're sure it's improving, and to measure anything you can that correlates with improvement.

i assume the agents are still doing MCTS; assuming you have a case where you know of a move that's better than chance, and you have a value-network that knows the results of a move is good, you should be able to see pi, the policy target, concentrate on that good move.  

Looking at the training outputs on tensorboard should show some useful things as well:  are losses going down?  Is the holdout vs the training data metrics similar or divergent? etc.

**Comment by MarkusMontenegro** [comment]
Hi, Thanks for the quick replies.
Im totally new to github, so im not sure if this is worth opening a new branch. Instead, here are all the files I have modified and the script I used to train multiple iteration. I added ".txt" to everything so I can upload it here.

[train.sh.txt](https://github.com/tensorflow/minigo/files/3959874/train.sh.txt)
[features.py.txt](https://github.com/tensorflow/minigo/files/3959875/features.py.txt)
[go.py.txt](https://github.com/tensorflow/minigo/files/3959876/go.py.txt)
[mcts.py.txt](https://github.com/tensorflow/minigo/files/3959877/mcts.py.txt)

About the question if there are cases where I know moves that are better than chance: When an agent is next to a point that needs to be collected (= increases own points when walking on it), the agents go in 3 of 4 cases somewhere else. Is there a way to have look  at the value-network?
I'll have a look at the training outputs too.
Thanks :)

**Comment by brilee** [comment]
Creating a commit and uploading it to a branch allows GitHub to render your diffs so that we can see what's changed at a glance - you should try it!

**Comment by amj** [comment]
@MarkusMontenegro even a value network is not needed; the question is the difference in the distribution of the policy target after search.

I.e., if there are 4 possible choices, a "flat" policy would be 25% on each.  If one of those choices results in a reward, you should expect to see that after search, that option should have > 25% of the visits.  As long as the actions that result in rewards gather "mass", things are working.

**Comment by MarkusMontenegro** [comment]
Ok, using your advices I found some stuff that might cause learning problems.
First of all, the position of the targets was not exported to the learning data, so learning the game correctly was theoretically impossible.
Second thing was the bahavior not to hard pick moves in the beginning of a game. In a situation, where the next point is about 10 fields away, there is not much difference between a move, that brings you 1 field closer, than to a move, that brings you 1 field away or to passing.  This caused the wrong moves to gather too much probability mass. Combined with the fact, that each move excludes all other moves in future (not the case in Go) lead to a suboptimal play style and even caused game overs verry often by passing twice in a row.
Thaks for the help and Merry Christmas!
